src/state/routing/useRoutingAPITrade.ts

```diff
@@ -45,17 +47,34 @@ export function useRoutingAPITrade<TTradeType extends TradeType>(
     routerPreference,
   })
 
+  const shouldUseRoutingApiV2 = useRoutingAPIV2Enabled()
+
   const {
-    isError,
-    data: tradeResult,
-    currentData: currentTradeResult,
-  } = useGetQuoteQuery(queryArgs ?? skipToken, {
+    isError: isLegacyAPIError,
+    data: legacyAPITradeResult,
+    currentData: currentLegacyAPITradeResult,
+  } = useGetQuoteQuery(!shouldUseRoutingApiV2 ? queryArgs ?? skipToken : skipToken, {
     // Price-fetching is informational and costly, so it's done less frequently.
     pollingInterval: routerPreference === INTERNAL_ROUTER_PREFERENCE_PRICE ? ms`1m` : AVERAGE_L1_BLOCK_TIME,
     // If latest quote from cache was fetched > 2m ago, instantly repoll for another instead of waiting for next poll period
     refetchOnMountOrArgChange: 2 * 60,
   })
 
+  const {
+    isError: isV2APIError,
+    data: v2TradeResult,
+    currentData: currentV2TradeResult,
+  } = useGetQuoteQueryV2(shouldUseRoutingApiV2 ? queryArgs ?? skipToken : skipToken, {
```

Comment:
this is the same as above but reversed? maybe it's one block to assign `v1QueryArg` and `v2QueryArg` above?

Reply:
would prefer to keep these separate for ease of deletion later